staging/src/k8s.io/apiserver/pkg/server/options/encryption_provider_config_test.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"strings"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/storage/value"
+)
+
+var correctConfig string = `
+- kind: k8s-aes-gcm
+  version: v1
+  keys:
+    - name: key1
```

Comment:
technically the identity transformer controls the prefix `{` and `k8s[\x00-\xFF]` already.  We need to prevent via validation people specifying that for any encryption provider except the default.